test/unit/test_shapetracker.py

```diff
@@ -227,6 +227,119 @@ def test_permute_reshape_2(self):
       new_st.append(st)
     self.sts = new_st
 
+  def test_reshaping_splitting(self):
+    self.st = CheckingShapeTracker((5,10,5,10))
+    self.st.permute((1, 0, 3, 2))
+    self.st.pad(((0,0), (0,5), (0,0), (0,5)))
+    self.st.reshape((10,2,5,10,2,5))
+    assert len(self.st.views) == 1
+    self.st.assert_same()
+
+  def test_reshape_splitting_1(self):
+    self.st = CheckingShapeTracker((1,10,1))
+    self.st.pad(((0,4),(0,0),(1,0)))
+    self.st.reshape((5,5,2,2))
+    assert len(self.st.views) == 1
+    self.st.assert_same()
+
+  def test_reshape_combining_1(self):
+    self.st = CheckingShapeTracker((2,1,10))
+    self.st.pad(((2,6), (0,0), (0,0)))
+    self.st.reshape((100,))
+    assert len(self.st.views) == 1
+    self.st.assert_same()
+
+  def test_reshape_combining_2(self):
+    self.st = CheckingShapeTracker((1,1,5))
+    self.st.pad(((3,6), (0,0), (0,5)))
+    self.st.reshape((100,))
+    assert len(self.st.views) == 1
+    self.st.assert_same()
+
+  def test_reshape_combining_3(self):
+    self.st = CheckingShapeTracker((1,1,4))
+    self.st.pad(((3,6), (0,0), (1,5)))
+    self.st.reshape((100,))
+    assert len(self.st.views) == 1
+    assert self.st.views[0].mask[0] == (31, 35)
+    self.st.assert_same()
+
+  def test_reshape_combining_4(self):
+    self.st = CheckingShapeTracker((1,1,5,5,1,1,5))
+    self.st.pad(((3,6), (0,0), (0,5), (0,0), (3,6), (0,0), (0,5)))
+    self.st.reshape((100,5,100))
+    assert len(self.st.views) == 1
+    self.st.assert_same()
+
+  def test_reshape_splitting_combining(self):
+    self.st = CheckingShapeTracker((1,5,5))
```

Comment:
this also works now!
Ill spend some more time in identifying cases where view.reshape returns None.